models/shopify__customer_email_cohorts.sql

```diff
@@ -0,0 +1,69 @@
+with calendar as (
+
+    select *
+    from {{ ref('shopify__calendar') }}
+    where cast({{ dbt.date_trunc('month','date_day') }} as date) = date_day
+
+), customers as (
+
+    select *
+    from {{ ref('shopify__customer_emails') }}
+
+), orders as (
+
+    select *
+    from {{ ref('shopify__orders') }}
+
+), customer_calendar as (
+
+    select
+        calendar.date_day as date_month,
+        customers.email,
+        customers.first_order_timestamp,
+        customers.source_relation,
+        {{ dbt.date_trunc('month', 'first_order_timestamp') }} as cohort_month
+    from calendar
+    inner join customers
+        on cast({{ dbt.date_trunc('month', 'first_order_timestamp') }} as date) <= calendar.date_day
+
+), orders_joined as (
+
+    select 
+        customer_calendar.date_month, 
+        customer_calendar.email, 
+        customer_calendar.first_order_timestamp,
+        customer_calendar.cohort_month,
+        customer_calendar.source_relation,
+        coalesce(count(distinct orders.order_id), 0) as order_count_in_month,
+        coalesce(sum(orders.order_adjusted_total), 0) as total_price_in_month,
+        coalesce(sum(orders.line_item_count), 0) as line_item_count_in_month
+    from customer_calendar
+    left join orders
+        on customer_calendar.email = orders.email
+        and customer_calendar.source_relation = orders.source_relation
+        and customer_calendar.date_month = cast({{ dbt.date_trunc('month', 'created_timestamp') }} as date)
+    group by 1,2,3,4,5
```

Comment:
similar question on group bys and dbt_utils

Reply:
this one i should use indeed group_by for